packages/ipfs-core-utils/src/with-timeout-option.js

```diff
@@ -0,0 +1,106 @@
+/* eslint-disable no-unreachable */
+'use strict'
+
+const TimeoutController = require('timeout-abort-controller')
+const { anySignal } = require('any-signal')
+const parseDuration = require('parse-duration').default
+const { TimeoutError } = require('./errors')
+
+/**
+ * @template {any[]} ARGS
+ * @template {Promise<any> | AsyncIterable<any>} R - The return type of `fn`
+ * @param {Fn<ARGS, R>} fn
```

Comment:
@achingbrain this seems to have introduced a regression, because `Fn` typedef ended up in `index.js` which is not imported here. Which for some reason TS still seems to kind of pick up, but it no longer behaves correctly & causing returned function to be inferred as any.

Reply:
I think the whole `Fn` type is kind of redundant and we should probably change type annotation for `withTimeoutOptions` to  something like this instead:

```js
/**
 * @template {any[]} Args
 * @template {Promise<any> | AsyncIterable<any>} R - The return type of `fn`
 * @param {(...args:Args) => R} fn
 * @param {number} [optionsArgIndex]
 * @returns {(...args:Args) => R}
 */
```